code/WorkableRestfulServiceFactory.php

```diff
@@ -1,48 +1,46 @@
 <?php
 
+namespace SilverStripe\Workable;
+
+use RuntimeException;
+use SilverStripe\Core\Environment;
+use SilverStripe\Workable\Workable;
+use SilverStripe\Core\Injector\Factory;
+
 /**
  * Factory to build a RestfulService object for workable. This is needed for three reasons:
  *
  * - It sets the $apiKey by preferring the constant, but falling back on the config layer
  * - RestfulService needs a URL in its constructor, and this is a computed value based on the
  *   static workable API base URL plus the user's subdomain
- * - It makes various boilerplate configurations to RestfulService, e.g. cache expiry, auth headers.
+ * - It makes various boilerplate configurations to RestfulService, e.g. auth headers.
  */
-class WorkableRestfulServiceFactory implements SilverStripe\Framework\Injector\Factory {
-
-	/**
-	 * Create the RestfulService (or whatever dependency you've injected)
-	 * @param  string $service 
-	 * @param  array  $params 
-	 * @return RestfulService
-	 */
-    public function create($service, array $params = []) {;
-    	$config = Workable::config();
-    	$subdomain = $config->subdomain;
+class WorkableRestfulServiceFactory implements Factory
+{
+    /**
+     * Create the RestfulService (or whatever dependency you've injected)
+     * @return RestfulService
+     */
+    public function create($service, array $params = [])
+    {
+        $apiKey = Environment::getEnv('WORKABLE_API_KEY');
```

Comment:
A bit nitpicky here, but this would be more testable if you added the key via the config `params`., rather than relying on the environment state, e.g.

```
apiKey: '`WORKABLE_API_KEY`'
```

Reply:
Do you have a working example of this? I can only either get the string set as the value or not at all.

Reply:
Yeah this one's pretty low priority, so don't let it be a blocker if it's too much work, but the idea is that factories themselves can have their constructors composed through injector. So:

```
SilverStripe\Core\Injector\Injector:
  SilverStripe\Workable\WorkableRestfulServiceFactory:
    constructor:
      key: '`WORKABLE_API_KEY`'
  SilverStripe\Workable\WorkableRestfulService:
    factory: SilverStripe\WorkableWorkableRestfulServiceFactory
  ```

Then your tests just become:

```php
$factory = new WorkableRestfulServiceFactory('123');
$factory->create(WorkableRestfulService::class);
```

Reply:
Awesome thanks, I'd gotten myself in a muddle mixing up properties and config and constructors and needed to be slapped with a wet fish.